client/src/routes/__layout.svelte

```diff
@@ -1,40 +1,111 @@
-<script>
-	import { goto } from '$app/navigation';
-	import { page } from '$app/stores';
+<script lang="ts">
+	import type { TopAppBarComponentDev } from '@smui/top-app-bar';
+
+	import { onMount } from 'svelte';
+
 	import LayoutGrid, { Cell } from '@smui/layout-grid';
 	import Button, { Label as ButtonLabel } from '@smui/button';
-	import { session } from '$app/stores';
+	import TopAppBar, { Row, Section } from '@smui/top-app-bar';
+	import IconButton from '@smui/icon-button';
+
+	import { goto } from '$app/navigation';
+	import { session, page } from '$app/stores';
 
 	import '../style/app.scss';
 
-	async function signout() {
-		await fetch('/api/logout', { method: 'POST' });
+	let darkTheme: boolean;
+	let topAppBar: TopAppBarComponentDev;
+
+	$: modeLabel = `Switch to ${darkTheme ? 'light' : 'dark'} mode`;
+	$: modeIcon = darkTheme ? 'light_mode' : 'dark_mode';
 
+	onMount(() => {
+		darkTheme = window.matchMedia('(prefers-color-scheme: dark)').matches;
+	});
+
+	const toggleMode = () => (darkTheme = !darkTheme);
+
+	const signout = async () => {
+		await fetch('/api/logout', { method: 'POST' });
 		location.reload();
-	}
+	};
+
+	const toHome = () => {
+		goto('/');
+	};
 
-	function toDashboard() {
+	const toProfile = () => {
+		goto('/profile');
+	};
+
+	const toDashboard = () => {
 		goto('/dashboard');
-	}
+	};
 
 	const isLoggedIn = () => {
 		return !!$session.user;
 	};
+
+	$: isCurrentPage = (pageName: string) => {
+		return $page.url.pathname === pageName;
+	};
 </script>
 
 <svelte:head>
 	<title>VeryRezsi</title>
+
```

Comment:
Nicely done, good to see a light/dark mode implementation, never have seen one myself. There is a strange flicker though sometimes on switch, but we'll see what causes it in the future.

Reply:
Do you mean, page content flickers a bit on switch? I have seen that.

 I think it is caused by switching entire stylesheets on-the-fly, and it flickers while the browser processes the new _smui_ stylesheet. We can dig a solution later, but I cannot see a simple fix.

Reply:
Yep that one, SMUI strikes again?